### ISSUE TITLE
Update-Help fails for WindowsUpdateProvider

### ISSUE THREAD

**Issue by joeyaiello** [MicrosoftDocs/windows-powershell-docs#139]
_From @dsolodow on February 9, 2018 15:44_

update-help : Failed to update Help for the module(s) 'WindowsUpdateProvider' with UI
culture(s) {en-US} : Unable to retrieve the HelpInfo XML file for UI culture en-US. Make sure the HelpInfoUri property
in the module manifest is valid or check your network connection and then try the command again.
At line:1 char:1
+ update-help
+ ~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [Update-Help], Exception
    + FullyQualifiedErrorId : UnableToRetrieveHelpInfoXml,Microsoft.PowerShell.Commands.UpdateHelpCommand

_Copied from original issue: PowerShell/PowerShell-Docs#2117_

**Comment by cebulloc** [comment]
I have the same error message. It has been going on for a couple of months now. Has anyone found a fix? or is there no update file for that particular module at the moment? I should mention I am completely new to PowerShell and IT in general. Any help would be appreciated. I am troubleshooting myself but I can not find anything related specifically related to "windows update provider." I have found some forums on other modules with a similar errors. 
Thank you for your assistance with this matter.

**Comment by dsolodow** [comment]
No, there isn't a fix for the missing help yet. However, most of the cmdlets in this module don't have any real options and just return a true/false or a date time.

**Comment by cebulloc** [comment]
dsolodow,
Thank you for your help. I appreciate you taking the time.

**Comment by Maverick5533** [comment]
I don't normally post any comments in GH, but I thought I would since Don Jones convinced me to become a "Master" (in the sense that if you learn something....share it). 

Let me preface my comments below that in this case I am strictly referring to the issues with Windows PowerShell 5.1 and other than the issue issue with updating the local Help file content for the WindowsUpdateProvider, all other help content was successfully updated. 

I also noticed on PowerShell Core 6.0 Prev, there is no module name WindowsUpdateProvider. My thought was to see if that module was available for Core 6.0 and then borrow the updated URI from the manifest in Core and then replace the URI in the manifest in Windows PowerShell 5.1 (make logical sense?). This was my first attempt in quickly resolving the issue (after skimming through other blogs online where other admins were running into this issue). 

Regarding the particular issue, I located the HelpURI  that is referenced in the manifest file for the WindowsUpdateProvider module. I then copied that URI into a browser which I suspect may redirect to a resource under docs.microsoft.com. 

Well long story short, it's a dead link (which is what POSH was trying to tell me in the console when it failed to update the help file). In summary, I suspect (just 'suspect'), at the moment, that not a whole lot of development efforts are allocated to update the module manifest for WindowsUpdateProvider given the fact that the evolutionary movement is toward POSH Core development. That said, I know folks can contact the specific MS Engineering groups and request edits/new features so that may be a option for resolving the bad URI in the WindowsUpdateProvider manifest.

**Comment by lmstearn** [comment]
Minor rantovani: Could the error message tell us a little more? Or at least, from the answer to [this thread](https://answers.microsoft.com/en-us/windows/forum/all/updateing-powershell-user-help-files/07afd880-c543-4e56-9446-1e9eb509003d) we know where the issue is, and quickly move on?
Thanks (violins al fine) :violin:

**Comment by 486DX50** [comment]
A year later this was not resolved

PS C:\Windows\system32> update-help
update-help : Failed to update Help for the module(s) 'WindowsUpdateProvider' with UI culture(s) {en-US} : Unable to retrieve the HelpInfo XML file for UI culture en-US. Make
sure the HelpInfoUri property in the module manifest is valid or check your network connection and then try the command again.
At line:1 char:1
+ update-help
+ ~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [Update-Help], Exception
    + FullyQualifiedErrorId : UnableToRetrieveHelpInfoXml,Microsoft.PowerShell.Commands.UpdateHelpCommand

Unbelievable...

**Comment by schtobia** [comment]
Same for de-DE:

    PS C:\WINDOWS\system32> Update-Help
    Update-Help : Fehler beim Aktualisieren von Hilfe für die Module "WindowsUpdateProvider" mit den Benutzeroberflächenkulturen {de-DE}: Die XML-Datei "HelpInfo" für die Benutzeroberflächenkultur de-DE kann nicht abgerufen werden. Stellen Sie sicher, dass die HelpInfoUri-Eigenschaft im Modulmanifest gültig
    ist, oder überprüfen Sie die Netzwerkverbindung, und führen Sie den Befehl dann erneut aus.
    In Zeile:1 Zeichen:1
    + Update-Help
    + ~~~~~~~~~~~
        + CategoryInfo          : ResourceUnavailable: (:) [Update-Help], Exception
        + FullyQualifiedErrorId : UnableToRetrieveHelpInfoXml,Microsoft.PowerShell.Commands.UpdateHelpCommand

**Comment by jeffreymcclain** [comment]
Same for me.

**Comment by junkiebev** [comment]
+1

**Comment by lifemanship** [comment]
+1

**Comment by ngetchell** [comment]
Ran into this today

**Comment by Dbones202** [comment]
same.

**Comment by SamB** [comment]
Well, [the Module Manifest for WindowsUpdateProvider](https://gist.github.com/SamB/da46329bf616f47faf5ce0224f166e46) on this machine doesn't give any attribution beyond "Microsoft Corporation", but it does specify a `HelpInfoUri` of <https://go.microsoft.com/fwlink/?linkid=390794>, so there's actually no need to futz around figuring out where the module comes from and somehow make the case that this warrants issuing a patch to WindowsUpdateProvider -- it's just necessary to get that redirect pointed somewhere appropriate.

**Comment by pdario** [comment]
Same here; I see that I haven't any *HelpInfo.xml in 
C:\Windows\System32\WindowsPowerShell\v1.0\Modules\WindowsUpdateProvider

Do you think this is correct??

Plus: do you know why I see more files in those dirs within CLI as compared to File Explore???

**Comment by junkiebev** [comment]
```
Error: 'read tcp 10.42.7.0:52124->10.42.5.9:8000: read: connection reset by peer'
Trying to reach: 'http://10.42.5.9:8000/index.js' 
```
is also a thing I've gotten

**Comment by rcl4444** [comment]
Same for me.
my host information below:
> Name             : ConsoleHost
> Version          : 5.1.18985.1
> InstanceId       : 7853b109-f755-4187-aafb-0bced2a37e95
> UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
> CurrentCulture   : en-US
> CurrentUICulture : en-US
> PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
> DebuggerEnabled  : True
> IsRunspacePushed : False
> Runspace         : System.Management.Automation.Runspaces.LocalRunspace

**Comment by gemzon** [comment]
My contribution of failed updates.

Searched for a solution (after a long time of failures which I've ignored).
The only thing found was several old issues, not been taken care of. Why not?

Also, read above about support suggesting a Windows re-installation. Poor guy.
```
Name                        Version   HelpInfoUri
----                        -------   -----------

- Response status code does not indicate success: 404 (Not Found).

PSDesiredStateConfiguration 2.0.4     https://go.microsoft.com/fwlink/?linkid=390814
WindowsUpdateProvider       1.0.0.2   https://go.microsoft.com/fwlink/?linkid=390794

- A Help URI cannot contain more than 10 redirections.

ConfigDefender              1.0       https://go.microsoft.com/fwlink/?linkid=390762
Defender                    1.0       https://go.microsoft.com/fwlink/?linkid=390762
HgsClient                   1.0.0.0   https://go.microsoft.com/fwlink/?linkid=519076
HgsDiagnostics              1.0.0.0   https://go.microsoft.com/fwlink/?LinkId=760404
UEV                         2.1.639.0 https://go.microsoft.com/fwlink/?LinkId=826061
Whea                        2.0.0.0   https://go.microsoft.com/fwlink/?linkid=390848
WindowsDeveloperLicense     1.0.0.0   https://go.microsoft.com/fwlink/?linkid=285578
```

**Comment by MysticRyuujin** [comment]
My results are the same as @gemzon after having installed PowerShell 7 RC1 and performing Update-Help

**Comment by josephyoung** [comment]
thank for @ScriptAutomate  's solution, this work around works: 

`Update-Help -Verbose -Force -ErrorAction SilentlyContinue`

**Comment by MysticRyuujin** [comment]
That's just ignoring the error. It's not a workaround unless you just don't want the error output.

**Comment by vazome** [comment]
How can this happen that they ignore this since the mass occurrence of the issue? 
Microsoft, ffs are you ~~doing~~?

**Comment by kodyabbott** [comment]
So frustrating!? How can this still be an issue YEARS later?

**Comment by PrMinisterGR** [comment]
```
PS C:\Windows\system32> $updates = Start-WUScan -SearchCriteria "IsInstalled=0 AND IsHidden=0 AND IsAssigned=1"
PS C:\Windows\system32> Install-WUUpdates -Updates $updates
Install-WUUpdates : Failed to install updates
At line:1 char:1
+ Install-WUUpdates -Updates $updates
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (MSFT_WUOperations:root/Microsoft/...FT_WUOperations) [Install-WUUpdates], CimException
    + FullyQualifiedErrorId : HRESULT 0x80240024,Install-WUUpdates
```


Without the help files is pretty hard to understand what's going on.

**Comment by MisotheisticYHWH** [comment]
> So frustrating!? How can this still be an issue YEARS later?

My guess is that not enough people are actually using the modules listed here and having issues, so this stays at a very low priority for the dev team: ConfigDefender, Defender, Microsoft.PowerShell.Operation.Validation, UEV, Whea, WindowsDeveloperLicense

But with this being open source, someone could tackle writing all of this documentation and fixing the URI, LOL =)

**Comment by PrMinisterGR** [comment]
It is working if you fo
`Install-WUUpdates -Updates (Start-WUScan Start-WUScan -SearchCriteria "IsInstalled=0 AND IsHidden=0 AND IsAssigned=1")`

The real issue is that it won't pull updates consistently, no driver updates, no supplemental updates, no matter how Start-WUScan is called, there is no way to completely automate this.
This is SO disappointing, every single Linux distro and BSD has solved this problem thirty years ago, Jesus Christ Microsoft.

**Comment by pdario** [comment]
This is the result (after deleting the repeated Start-WUScan):

Install-WUUpdates -Updates (Start-WUScan -SearchCriteria "IsInstalled=0 AND IsHidden=0 AND IsAssigned=1")
Invoke-CimMethod : WU Scan Failed
At C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\WindowsUpdateProvider\MSFT_WUOperations.psm1:17 char:20
+ ...  $scanres = Invoke-CimMethod -Namespace root/Microsoft/Windows/Window ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (root/Microsoft/...FT_WUOperations:String) [Invoke-CimMethod], CimException
    + FullyQualifiedErrorId : HRESULT 0x80240438,Microsoft.Management.Infrastructure.CimCmdlets.InvokeCimMethodCommand
 
Start-WUScan : Scan hit error: .ReturnValue
At line:1 char:29
+ ... s -Updates (Start-WUScan -SearchCriteria "IsInstalled=0 AND IsHidden= ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Start-WUScan
 
Install-WUUpdates : Cannot validate argument on parameter 'Updates'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
At line:1 char:28
+ ... es -Updates (Start-WUScan -SearchCriteria "IsInstalled=0 AND IsHidden ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Install-WUUpdates], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Install-WUUpdates

**Comment by PrMinisterGR** [comment]
Right, I figured it out.
You need to run
`Start-WUScan` by itself first, let it finish.

And then you call it with `Install-WUUpdates -Updates (Start-WUScan)`
<img width="800" alt="image" src="https://user-images.githubusercontent.com/12447314/78767740-44ce3080-798b-11ea-9157-eeff295ec2cb.png">

**Comment by pdario** [comment]
No good

![image](https://user-images.githubusercontent.com/5813131/78769906-36354880-798e-11ea-8bd8-e8b803e346f9.png)

**Comment by PrMinisterGR** [comment]
This is with an Admin Powershell, I presume, correct?

**Comment by pdario** [comment]
Yes; it starts in C:\Windows\System32 when it is elevated.

Win ver:

![image](https://user-images.githubusercontent.com/5813131/78781533-bb295d80-79a0-11ea-9166-9ed831f70d41.png)

**Comment by felixkasza** [comment]
Please supply at least some dummy help. Pretty much the first thing a user will do upon installing Powershell is looking at the help -- BOOM! Big, fat errors. It's been _years_, surely you can find someone to write "There is no help for this cmdlet"?

**Comment by PrMinisterGR** [comment]
> 
> 
> Yes; it starts in C:\Windows\System32 when it is elevated.
> 
> Win ver:
> 
> ![image](https://user-images.githubusercontent.com/5813131/78781533-bb295d80-79a0-11ea-9166-9ed831f70d41.png)

I don't think I can help much more, I am as much in the dark as you are guys :(

The largest software development company in the world, and it can't write a system updated better than any run of the mill Linux distro. Shame.

**Comment by jonathanaulson** [comment]
My first foray into powershell.  I'm impressed.  These engineers are bad mf'ers if they get away with this at work.  Respect.

**Comment by MarkIannucci** [comment]
@PrMinisterGR , we're avoiding a second call to `Start-WUScan` by storing the `Start-WUScan` output in a variable.  This seems to be more? reliable at least for our configuration.  Full code follows:

````
$updates = Start-WUScan
Install-WUUpdates -Updates $updates
````

**Comment by PrMinisterGR** [comment]
These tools are no longer integrated at all in Windows 10 2004.

**Comment by ScriptAutomate** [comment]
It looks like most modules are now properly getting their updates, due to fixes done to either `Update-Help` or to the `HelpInfoURI` endpoints in many built-in/default Microsoft PowerShell modules for Windows.

As for the `WindowsUpdateProvider` module that started this large issue discussion around updating modules in general, it looks like they have simply _removed_ the `HelpInfoURI` for it so that it no longer errors on update because _it has no endpoint where updates would exist (that I am aware of)._

So, technically, the problem that started this entire issue has been resolved: _it no longer fails._

> These tools are no longer integrated at all in Windows 10 2004.

@PrMinisterGR Do you mean the commands in the `WindowsUpdateProvider` module, specifically?

I did do a quick test, and I get a `Provider load failure` error having to do with `MSFT_WUOperations` by the looks of it. But this was with a quick test, and I currently don't have the patience to try and resolve the issue following workaround directions similar to this:

- [Fixing a WMI provider load failure](https://p0w3rsh3ll.wordpress.com/2014/12/28/fixing-a-wmi-provider-load-failure/)

This is with Windows 10 20H2 (newer than 2004, which was 20H1).

**Comment by KingWAR10CK** [comment]
Im still getting the same fault:
![image](https://user-images.githubusercontent.com/17877055/88583724-ebc8fc00-d050-11ea-9b60-b035bbfef0f2.png)

**Comment by ScriptAutomate** [comment]
> **NOTE:** The following is tested with Windows 10 20H2 (newer than 2004, which was 20H1), seen as build `19042`.
>
> ```
> PS > Get-CimInstance win32_operatingsystem | select Caption,Version
> 
> Caption                  Version
> -------                  -------
> Microsoft Windows 10 Pro 10.0.19042
> ```

@KingWAR10CK I checked again, and was definitely mistaken. I forgot I had commented out the `HelpInfoURI` in order to avoid the errors!

```
PS > update-help                                                                                                                                     update-help : Failed to update Help for the module(s) 'ConfigDefender, HostNetworkingService, PSReadline, WindowsUpdateProvider' with UI culture(s) {en-US} : Unable to retrieve the HelpInfo
XML file for UI culture en-US. Make sure the HelpInfoUri property in the module manifest is valid or check your network connection and then try the command again.
At line:1 char:1
+ update-help
+ ~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [Update-Help], Exception
    + FullyQualifiedErrorId : UnableToRetrieveHelpInfoXml,Microsoft.PowerShell.Commands.UpdateHelpCommand

update-help : Failed to update Help for the module(s) 'HgsClient, HgsDiagnostics' with UI culture(s) {en-US} : Unable to connect to Help content. The server on which
Help content is stored might not be available. Verify that the server is available, or wait until the server is back online, and then try the command again.
At line:1 char:1
+ update-help
+ ~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Update-Help], Exception
    + FullyQualifiedErrorId : UnableToConnect,Microsoft.PowerShell.Commands.UpdateHelpCommand
```

Broken help in Windows 10 Pro 20H2:
- ConfigDefender
- HostNetworkingService
- PSReadline
- **WindowsUpdateProvider**
- HgsClient
- HgsDiagnostics

**Comment by ScriptAutomate** [comment]
For those interested, https://github.com/PowerShell/PowerShell/issues/13591 is looking at the same problem, but from the perspective of modifying PowerShell's behavior. I suggest taking a look and providing input :)

**Comment by ITMicaH** [comment]
I'm running Windows 10 20H1 and 20H2 and I can't find the WindowsUpdateProvider module anymore. I had it and used it in 19H2 but now it's gone. Copying it from a 1909 desktop won't work either because the CIM namespace root/Microsoft/Windows/WindowsUpdate isn't there anymore. Without documentation I have no clue how to get it back.

**Comment by sdwheeler** [comment]
The commentary in this issue has diverged widely from the original problem report about Update-Help. The Update-Help issues have already been reported many times. We are in the process of fixing these issues but it will be a few months before we can implement the final fix.

The other issues with specific cmdlet behavior should be reported as product issues at: https://windowsserver.uservoice.com/forums/301869-powershell.

Documentation issues about specific cmdlets should be added to the repository with separate issues for each cmdlet.

**Comment by ScriptAutomate** [comment]
> The commentary in this issue has diverged widely from the original problem report about Update-Help. The Update-Help issues have already been reported many times. We are in the process of fixing these issues but it will be a few months before we can implement the final fix.
> 
> The other issues with specific cmdlet behavior should be reported as product issues at: https://windowsserver.uservoice.com/forums/301869-powershell.
> 
> Documentation issues about specific cmdlets should be added to the repository with separate issues for each cmdlet.

@sdwheeler UserVoice is no longer used by Microsoft. Do you know where users may be able to raise issues regarding built-into-Windows PowerShell modules, that don't have public repos, such as the other ones listed here (outside of `PSReadLine`, due to your blog entries that address how to fix):

- https://devblogs.microsoft.com/powershell/updating-help-for-the-psreadline-module/
- https://devblogs.microsoft.com/powershell/updating-help-for-the-psreadline-module-in-windows-powershell-5-1/

I was able to find this page here, [Microsoft Support: UserVoice Pages](https://support.microsoft.com/en-us/office/uservoice-pages-430e1a78-e016-472a-a10f-dc2a3df3450a):

> **Note:** We will be moving away from UserVoice feedback sites on a product-by-product basis throughout the 2021 calendar year. We will leverage 1st party solutions for customer feedback.
>
> **How can I submit feedback to Microsoft?**
> We have a variety of channels that allow customers to communicate with us. Including:
> - In-product experiences
> - Windows Feedback Hub
> - [Microsoft Tech Community](https://techcommunity.microsoft.com/)
> - Microsoft Store

It looks like the tech community space that may be most relevant is [Microsoft Tech Community: PowerShell](https://techcommunity.microsoft.com/t5/powershell/ct-p/WindowsPowerShell). Would this then be the best place to raise issues and bring attention to internally-sourced PowerShell modules, broken `HelpInfoURI` functionality, etc?

**Comment by sdwheeler** [comment]
@ScriptAutomate In response to your final point, you don't need to bring attention to  the internally-sourced PowerShell modules. We are well aware of the problems and are working to fix them. We have fixed this (for 90%) of the modules for the Server 2022 and Windows 11 release. The harder part is to fix this for old versions of Windows. We hope to have fixed this by the end of the calendar year.

**Comment by ScriptAutomate** [comment]
@sdwheeler Awesome! Thanks for the info, for the blog posts, and for the other work I've seen you involved in across discussions/issues :) It's appreciated, and looks to be quite a lot of wrangling

**Comment by whitTech** [comment]
Can @ScriptAutomate, @sdwheeler, or anyone else on this thread provide a working link to track the progress of the WindowsUpdateProvider module being unavailable in newer versions of Windows? Suggestions have been provided, but they are either dead links or not relevant. 

This thread is linked in one of the top hits on Google: https://docs.microsoft.com/en-us/answers/questions/212035/start-wuscan-cmdlet-is-not-avialble-to-run-on-few.html

To avoid further off-topic posting here, any help with a legitimate destination would be beneficial for all.

**Comment by nebb00** [comment]
> Can @ScriptAutomate, @sdwheeler, or anyone else on this thread provide a working link to track the progress of the WindowsUpdateProvider module being unavailable in newer versions of Windows? Suggestions have been provided, but they are either dead links or not relevant.
> 
> This thread is linked in one of the top hits on Google: https://docs.microsoft.com/en-us/answers/questions/212035/start-wuscan-cmdlet-is-not-avialble-to-run-on-few.html
> 
> To avoid further off-topic posting here, any help with a legitimate destination for seekers would be beneficial for all and greatly appreciated.

please bring back the "WindowsUpdateProvider" module

**Comment by nebb00** [comment]
A third party module PSWindowsUpdate is not a great replacement for basic inbuilt functionality

**Comment by kevstar0** [comment]
A year later & still not fixed
running windows 11 pro
 version 21H2

**Comment by PrMinisterGR** [comment]
> @ScriptAutomate In response to your final point, you don't need to bring attention to the internally-sourced PowerShell modules. We are well aware of the problems and are working to fix them. We have fixed this (for 90%) of the modules for the Server 2022 and Windows 11 release. The harder part is to fix this for old versions of Windows. We hope to have fixed this by the end of the calendar year.

Hi, any possible update on this specific module? It would seem it would be important to be able to update the operating system via the command line, in a consistent way without 3rd party libraries.

**Comment by nebb00** [comment]
> A year later & still not fixed running windows 11 pro version 21H2

I actually made a powershell script to update hosts without the use of any modules. If I get time, I will remove any system specific hard coded items and publish on my GitHub.

Basically reads AD and gives you a list of enabled windows machines, allows you a choice. And updates and reboots the machine remotely.

**Comment by PrMinisterGR** [comment]
> > A year later & still not fixed running windows 11 pro version 21H2
> 
> I actually made a powershell script to update hosts without the use of any modules. If I get time, I will remove any system specific hard coded items and publish on my GitHub.
> 
> Basically reads AD and gives you a list of enabled windows machines, allows you a choice. And updates and reboots the machine remotely.

Hi, does this do optional updates? Can you please post what you have working?

**Comment by BWMerlin** [comment]
> > @ScriptAutomate In response to your final point, you don't need to bring attention to the internally-sourced PowerShell modules. We are well aware of the problems and are working to fix them. We have fixed this (for 90%) of the modules for the Server 2022 and Windows 11 release. The harder part is to fix this for old versions of Windows. We hope to have fixed this by the end of the calendar year.
> 
> Hi, any possible update on this specific module for Windows 11? It would seem it would be important to be able to update the operating system via the command line, in a consistent way without 3rd party libraries.

This does not update the operating system but rather updates the PowerShell help files.

**Comment by walthar** [comment]
Has anyone gotten any solution to the original problem? 

```
> update-help -Module ConfigDefender -force
update-help : Failed to update Help for the module(s) 'ConfigDefender' with UI culture(s) {fi-FI} : Unable to retrieve
the HelpInfo XML file for UI culture fi-FI. Make sure the HelpInfoUri property in the module manifest is valid or check
 your network connection and then try the command again.
At line:1 char:1
+ update-help -Module ConfigDefender -force
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [Update-Help], Exception
    + FullyQualifiedErrorId : UnableToRetrieveHelpInfoXml,Microsoft.PowerShell.Commands.UpdateHelpCommand


```

My system:
```
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.19044 N/A Build 19044
```

**Comment by sdwheeler** [comment]
@walthar The problem is in the module listed. That module is not owned by the PowerShell team. It is a Windows component. For modules that ship in Windows, use the Windows Feedback Hub. For more information, see [Send feedback to Microsoft with the Feedback Hub app][1].

[1]: https://support.microsoft.com/windows/send-feedback-to-microsoft-with-the-feedback-hub-app-f59187f8-8739-22d6-ba93-f66612949332

**Comment by walthar** [comment]
Thanks @sdwheeler . Will do. 
I do not intent to sound harsh, just from customer point of view powershell-team should be the single contact point for issues and they should assign those  internally amongst different teams. My limited view is that the issue presents itself in powershell, thus it is a powershell problem.
Ofc I wouldn't report eg vmware module issues here, but to vmware.

**Comment by MiladZarour** [comment]
I am having this error 

![image](https://user-images.githubusercontent.com/50852900/213278361-6ebb62b3-791e-4a6c-9b11-dd5c2f4e0093.png)

**Comment by sdwheeler** [comment]
@MiladZarour These are all known issues for those modules. The module owners have to fix those.

**Comment by merlinuwe** [comment]
Same error.

![image](https://user-images.githubusercontent.com/4138869/217618981-797d7acb-22a4-48f7-a747-acb92a1eeb95.png)

**Comment by WuJieFly** [comment]
thank for @ScriptAutomate 's solution, this work around works:

Update-Help -Verbose -Force -ErrorAction SilentlyContinue

**Comment by pdario** [comment]
It looks you're just silencing the error messages

**Comment by afberendsen** [comment]
> It looks like you're just silencing the error messages

Could be. But, at least, now I have all the help up-to-date. More likely, with the arguments, the update runs and do not stop under any circumstances. It is better to have something up-to-date then the update stopping becasue of an error.

**Comment by pdario** [comment]
I understand that the mutes error messages refer to help files not getting updated. Are you sure you're all updated now?

**Comment by browhynott** [comment]
This does not solve the problem I agree. Any real help on this topic maybe? Such a shame really we still have no proper answer.

**Comment by sdwheeler** [comment]
It's closed because it can't be fixed by the docs team or in this docs repository. It has to be fixed by the module owner in the Windows source code.

**Comment by iTZBALAM** [comment]
Im using Spanish language in my Computer. I was getting error messages when trying to update help. I decided to try with UICulture = en-US. I solve this issue running **Update-Help -UICulture en-US**

Now I can use Get-Help without issues
